language/vm/src/file_format.rs

```diff
@@ -1177,12 +1212,24 @@ pub enum Bytecode {
     ///
     /// ```..., -> ..., bytearray_value```
     GetTxnPublicKey,
+    /// Shift the (second top value) left (top value) bits and pushes the result on the stack.
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., u64_value(1), u64_value(2) -> ..., u64_value```
```

Comment:
That's sth I and @dariorussi have been discussing. No conclusions yet, need to look into it more.

Reply:
Hmm... Good catch. Looks like I forgot to update the comments. 
The correct stack transition is `[..., T1, T2] -> T1`, where `T1, T2` are integer types.

Your question still applies -- that is whether we should replace `T2` with `u8`. One concern I have with the `u8` approach is that is may require an extra cast instruction at certain places. I can see how `T2` being polymorphic complicates the implementation/language a little bit though.

Reply:
>  One concern I have with the u8 approach is that is may require an extra cast instruction at certain places

When will this happen with the current types?

> I can see how T2 being polymorphic complicates the implementation/language a little bit though.

Definitely; requiring T2 to be (e.g.) a u128 when shifting a 128 seems very silly.

Reply:
The scenario in my mind was that someone would end up having two integers of the same type, and then in order to perform a shift, the second integer needs to be converted to a u8 explicitly, by inserting a `CastU8`.

Hmm... maybe this is actually a good thing? Since the user is using the integer to represent the number of bits shifted, it would be a waste to use u64 or u128 in the first place.

Reply:
As discussed offline, I think this is indeed a good thing.

Reply:
Yeah I think we all prefer u8. I'll land this PR as is, but will put up a follow up PR to change it to u8.